crafters/nlp/PDFExtractorSegmenter/__init__.py

```diff
@@ -0,0 +1,57 @@
+from typing import Dict, List
+
+from jina.executors.crafters import BaseSegmenter
+import io
+
+
+class PDFExtractorSegmenter(BaseSegmenter):
+    """
+    :class:`PDFExtractorSegmenter` Extracts data (text and images) from PDF.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def craft(self, uri: str, buffer: bytes, *args, **kwargs) -> List[Dict]:
+        import fitz
+        import PyPDF2
+
+        if uri:
+            pdf_obj = open(uri, 'rb')
+        elif buffer:
+            pdf_obj = io.BytesIO(buffer)
+        else:
+            raise ValueError('No value found in "buffer" and "uri"')
+
+        #Extract images
+        imgs = []
+        pdf_doc = fitz.open(uri)
+        for i in range(len(pdf_doc)):
+            for img in pdf_doc.getPageImageList(i):
+                xref = img[0]
+                pix = fitz.Pixmap(pdf_doc, xref)
+                if pix.n - pix.alpha < 4:   # if gray or RGB
+                    pix.writePNG("p%s-%s.png" % (i, xref)) #Format is page, and image
```

Comment:
is this `writePNG` needed? won't it create local images?

Reply:
But this is the one that is actually creating the PNG from the pdf, is not that what we want?

Reply:
We do not want to dump it to disk